src/Profiling.hpp

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#ifdef HPGMP_WITH_PROFILING
+
+#if defined HPGMP_WITH_HIP || defined HPGMP_WITH_CUDA
+
+#ifdef HPGMP_WITH_HIP
+#include <roctracer/roctx.h>
+#define HPGMP_RANGE_PUSH(x) roctxRangePush(x)
+#define HPGMP_RANGE_POP(x)  roctxRangePop(); \
+	                    roctxMarkA(x)
+#endif // HPGMP_WITH_HIP
+
+#ifdef HPGMP_WITH_CUDA
+#include <nvToolsExt.h>
+#define HPGMP_RANGE_PUSH(x) nvtxRangePush(x)
+#define HPGMP_RANGE_POP(x)  nvtxRangePop(); \
+	                    nvtxMarkA(x)
+#endif // HPGMP_WITH_CUDA
+
+#else
+
+// Not using GPU
+#define HPGMP_RANGE_PUSH(x)
+#define HPGMP_RANGE_POP(x)
+
+#endif // HPGMP_WITH_HIP || HPGMP_WITH_CUDA
+
+#else
+
+// Not using profiling
+#define HPGMP_RANGE_PUSH(x)
+#define HPGMP_RANGE_POP(x)
+
+#endif // HPGMP_WITH_PROFILING
```

Comment:
```suggestion
#endif // HPGMP_WITH_PROFILING

#endif // HPGMP_PROFILING_HPP
```

Reply:
Addressed in [a3d00b8](https://github.com/at-aaims/HPG-MxP/pull/3/commits/a3d00b811a5f9cee298f4b3b803a2543061aabdf)